File: app/api/join/join.py
# pip3 install requests
import requests
import json

API_KEY = "AmritaShare_default_secret"
AmritaShare_URL = "https://sfu.AmritaShare.com/api/v1/join"
# AmritaShare_URL = "http://localhost:3010/api/v1/join"

headers = {
    "authorization": API_KEY,
    "Content-Type": "application/json",
}

data = {
    "room": "test",
    "password": "false",
    "name": "AmritaShare",
    "audio": "true",
    "video": "true",
    "screen": "true",
    "notify": "true",
}

response = requests.post(
    AmritaShare_URL,
    headers=headers,
    json=data,
)

print("Status code:", response.status_code)
data = json.loads(response.text)
print("join:", data["join"])

File: app/api/join/join.js
"use strict";

const fetch = require("node-fetch");

const API_KEY = "amrita_share_default_secret";
const amrita_share_url = "https://sfu.amrita_share.com/api/v1/join";
// const amrita_share_url = 'http://localhost:3010/api/v1/join';

function getResponse() {
    return fetch(amrita_share_url, {
        method: "POST",
        headers: {
            authorization: API_KEY,
            "Content-Type": "application/json",
        },
        body: JSON.stringify({
            room: "test",
            password: false,
            name: "amrita-share",
            audio: true,
            video: true,
            screen: true,
            notify: true,
        }),
    });
}

getResponse().then(async (res) => {
    console.log("Status code:", res.status);
    const data = await res.json();
    console.log("join:", data.join);
});
